pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1281,13 +1281,46 @@ func (self *AWSCloud) TCPLoadBalancerExists(name, region string) (bool, error) {
 	return false, nil
 }
 
+// Retrieves instance's vpc id from metadata
+func (self *AWSCloud) findVPCId() (string, error) {
```

Comment:
This could maybe be called findVPCFromMetadata (or maybe could just be the body of findVPC, as I propose that you don't keep the magic hard-coded name.)

Note that our style says that we write acronyms in consistent casing (I believe), and ID is considered an acronym, so e.g. vpcID, getID, findVPCID.  I find the last one particularly awkward, but that's our style!